bin/geni-sync-wireless

```diff
@@ -0,0 +1,618 @@
+                break
+
+        if not holdingpen_admin_info:
+            self.error("Holdingpen admin not in GENI: %s" % \
+                           self._options.holdingpen_admin)
+
+        # Grab 'pretty name' for holdingpen admin
+        admin_pretty_name = self.get_pretty_name(holdingpen_admin_info)
+        holdingpen_admin_username = \
+            self.to_orbit_name(self._options.holdingpen_admin)
+        holdingpen_admin_ssh_keys = holdingpen_admin_info['ssh_keys']
+
+        ldif_text = ""
+        if self._options.holdingpen_group not in self._orbit_groups:
+            ldif_text = ldif_text + \
+                self._orb.ldif_for_group(self._options.holdingpen_group,
+                                   self.holdingpen_group_description)
+            ldif_text = ldif_text + \
+                self._orb.ldif_for_group_admin(self._options.holdingpen_group,
+                                         holdingpen_admin_username,
+                                         self._options.holdingpen_group)
+            print "Creating holdingpen group: %s" % \
+                self._options.holdingpen_group
+
+        if holdingpen_admin_username not in self._orbit_users:
+            ldif_text = ldif_text + \
+                self._orb.ldif_for_user(holdingpen_admin_username,
+                                  self._options.holdingpen_group,
+                                  admin_pretty_name,
+                                  holdingpen_admin_info['first_name'],
+                                  holdingpen_admin_info['email_address'],
+                                  holdingpen_admin_info['last_name'],
+                                  holdingpen_admin_ssh_keys,
+                                  self.holdingpen_group_description, None)
+            print "Creating holdingpen admin: %s" % \
+                holdingpen_admin_username
+
+        if ldif_text != "":
+            self._orb.saveUser(ldif_text)
+
+    # Make sure that all members of wimax-enabled projects exist in orbit
+    # If not, create and place in holdingpen group as their primary group
+    # The holdingpen admin is in the list of geni members, but don't need
+    #   to create his account: should already be there
+    def ensure_project_members_exist(self): 
+        for member_id, member_info in self._geni_members.items():
+            username = member_info['username']
+            if username == self._options.holdingpen_admin: continue
+            orbit_username = self.to_orbit_name(username)
+            if orbit_username not in self._orbit_users:
+                member_pretty_name = self.get_pretty_name(member_info)
+                member_ssh_keys = member_info['ssh_keys']
+                if len(member_ssh_keys) == 0:
+                    print "Skipping user with no ssh keys: %s" % username
+                    continue
+                print "Creating ORBIT user: %s" % orbit_username
+                first_name = ""
```

Comment:
According to the old wimax-enable code, first_name must not be empty. So if there is not first_name attribute, the old code used the email address.